lib/lib-table-cluster/src/main/java/org/sagebionetworks/table/cluster/SQLTranslatorUtils.java

```diff
@@ -301,7 +304,15 @@ public static Row readRow(ResultSet rs, boolean includesRowIdAndVersion, boolean
 	 */
 	public static void translateModel(QuerySpecification transformedModel,
 			Map<String, Object> parameters,
-		  ColumnTranslationReferenceLookup columnTranslationReferenceLookup) {
+		  ColumnTranslationReferenceLookup columnTranslationReferenceLookup, Long userId) {
+
+		// Insert userId if needed
+		Iterable<NumericValueFunction> hasUser = transformedModel.createIterable(NumericValueFunction.class);
+		for(NumericValueFunction pred: hasUser){
+			if(pred.getChild() instanceof CurrentUserFunction){
+				pred.replaceChildren(new UnsignedLiteral(new UnsignedNumericLiteral(new ExactNumericLiteral(userId))));

```

Comment:
Please add a comment that the chain of UnsignedLiteral(unisgnedNumericLiteral... etc is needed for the binding of variables.